### PR TITLE
Start opening up the HTTP/2 types for http4s proper consumption

### PR DESCRIPTION
**Pull request by bryce-anderson**
We made a lot of the API private, but some of it should be made public
again for consumption by http4s proper.

### CODE REVIEW

**Comment by bryce-anderson** [comment]
@ChristopherDavenport, that's a good point. The real value is in the pseudo-header names. I'll move them into a separate object that is public and leave `StageTools` the way it is.

**Comment by bryce-anderson** [comment]
@ChristopherDavenport, I think this is better now if you care to take a second look.

File: examples/src/main/scala/org/http4s/blaze/examples/ExampleService.scala [hunk 1]
```diff
@@ -5,8 +5,7 @@ import java.nio.charset.StandardCharsets
 import java.util.concurrent.atomic.AtomicReference
 
 import org.http4s.blaze.channel.ServerChannel
-import org.http4s.blaze.http.util.HeaderNames
-import org.http4s.blaze.http._
+import org.http4s.blaze.http.{HeaderNames, _}
```
Comment (rossabaker) [inline examples/src/main/scala/org/http4s/blaze/examples/ExampleService.scala]: Uh, doesn't the underscore cover this?

File: http/src/main/scala/org/http4s/blaze/http/http2/server/RequestParser.scala [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 
-import org.http4s.blaze.http.http2.StageTools._
 import org.http4s.blaze.http.{BodyReader, HttpRequest, _}
 
 import scala.collection.mutable.ArrayBuffer
 
 /** Tool for turning a HEADERS frame into a request */
 private object RequestParser {
+  import org.http4s.blaze.http.http2.PseudoHeaders._
+  import org.http4s.blaze.http.http2.StageTools._
```
Comment (rossabaker) [inline http/src/main/scala/org/http4s/blaze/http/http2/server/RequestParser.scala]: Any reason not to put these with the rest of the imports?

File: examples/src/main/scala/org/http4s/blaze/examples/ExampleService.scala (hunk 1 above)
Comment (bryce-anderson) [reply examples/src/main/scala/org/http4s/blaze/examples/ExampleService.scala]: Yeah. IntelliJ rename fail.

File: http/src/main/scala/org/http4s/blaze/http/http2/server/RequestParser.scala (hunk 2 above)
Comment (bryce-anderson) [reply http/src/main/scala/org/http4s/blaze/http/http2/server/RequestParser.scala]: Personal preference I suppose. I'll move it.

**Pull request merged**